cpp/src/arrow/util/cpu_info.cc

```diff
@@ -288,7 +288,7 @@ bool RetrieveCPUInfo(int64_t* hardware_flags, std::string* model_name,
   if (features_ECX[19]) *hardware_flags |= CpuInfo::SSE4_1;
   if (features_ECX[20]) *hardware_flags |= CpuInfo::SSE4_2;
   if (features_ECX[23]) *hardware_flags |= CpuInfo::POPCNT;
-  if (features_ECX[23]) *hardware_flags |= CpuInfo::AVX;
+  if (features_ECX[28]) *hardware_flags |= CpuInfo::AVX;
```

Comment:
Thanks for this fix. Based on below doc, we did have a typo here.
https://en.wikipedia.org/wiki/CPUID#EAX=1:_Processor_Info_and_Feature_Bits